src/server/network/protocol/protocolgame.cpp

```diff
@@ -95,6 +95,112 @@ namespace {
+					ImbuementInfo imbuementInfo;
+					if (!weapon->getImbuementInfo(i, &imbuementInfo)) {
+						continue;
+					}
+
+					if (imbuementInfo.duration > 0) {
+						auto imbuement = *imbuementInfo.imbuement;
+						if (imbuement.combatType != COMBAT_NONE) {
+							msg.addByte(static_cast<uint32_t>(imbuement.elementDamage));
+							msg.addByte(getCipbiaElement(imbuement.combatType));
+							imbueDmg = true;
+							break;
+						}
+					}
+				}
+			}
+		}
+		if (!imbueDmg) {
+			msg.addByte(0);
+			msg.addByte(CIPBIA_ELEMENTAL_UNDEFINED);
+		}
+	}
+
+	/**
+	 * @brief Calculates the absorb values for different combat types based on player's equipped items.
+	 *
+	 * This function calculates the absorb values for each combat type based on the items equipped by the player.
+	 * The calculated absorb values are stored in the provided array.
+	 *
+	 * @param[in] player The pointer to the player whose equipped items are considered.
+	 */
+	void calculateAbsorbValues(Player* player, NetworkMessage &msg, uint8_t &combats) {
+		alignas(16) uint16_t damageReduction[COMBAT_COUNT] = { 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100 };
+
+		for (int32_t slot = CONST_SLOT_FIRST; slot <= CONST_SLOT_LAST; ++slot) {
+			if (!player->isItemAbilityEnabled(static_cast<Slots_t>(slot))) {
+				continue;
+			}
+
+			Item* item = player->getInventoryItem(static_cast<Slots_t>(slot));
+			if (!item) {
+				continue;
+			}
+
+			const ItemType &itemType = Item::items[item->getID()];
+			if (!itemType.abilities) {
+				continue;
+			}
+
+			for (uint16_t i = 0; i < COMBAT_COUNT; ++i) {
+				damageReduction[i] *= (std::floor(100 - itemType.abilities->absorbPercent[i]) / 100.);
+			}
+
+			uint8_t imbuementSlots = itemType.imbuementSlot;
+			if (imbuementSlots > 0) {
+				for (uint8_t slot = 0; slot < imbuementSlots; ++slot) {
+					ImbuementInfo imbuementInfo;
+					if (!item->getImbuementInfo(slot, &imbuementInfo)) {
+						continue;
+					}
+
+					if (imbuementInfo.duration > 0) {
+						auto imbuement = *imbuementInfo.imbuement;
+						for (uint16_t combat = 0; combat < COMBAT_COUNT; ++combat) {
+							const int16_t &imbuementAbsorbPercent = imbuement.absorbPercent[combat];
+
+							if (imbuementAbsorbPercent != 0) {
+								damageReduction[combat] *= (std::floor(100 - imbuementAbsorbPercent) / 100.);
+							}
+						}
+					}
+				}
+			}
+		}
+
+		for (size_t i = 0; i < COMBAT_COUNT; ++i) {
+			if (damageReduction[i] != 100) {
+				// Wheel of destiny resistance
+				damageReduction[i] += static_cast<int16_t>(player->wheel()->getResistance(indexToCombatType(i))) / 100.f;
+				if (g_configManager().getBoolean(TOGGLE_WHEELSYSTEM)) {
+					damageReduction[i] += static_cast<int16_t>(player->wheel()->getResistance(indexToCombatType(i))) / 100.f;
+				}
+
+				msg.addByte(getCipbiaElement(indexToCombatType(i)));
+				msg.addByte(std::max<int16_t>(-100, std::min<int16_t>(100, (100 - damageReduction[i]))));
+				++combats;
+			}
+		}
```

Comment:
Logic here changed quite a bit, this was resulting in negative values now. Plus the wheel logic was duplicated.
```suggestion
		for (size_t i = 0; i < COMBAT_COUNT; ++i) {
			damageReduction[i] -= player->getAbsorbPercent(indexToCombatType(i));
			if (g_configManager().getBoolean(TOGGLE_WHEELSYSTEM)) {
				damageReduction[i] -= static_cast<int16_t>(player->wheel()->getResistance(indexToCombatType(i))) / 100.f;
			}
			if (damageReduction[i] != 100) {
				spdlog::info("CombatType: {}, DamageReduction: {}", i, damageReduction[i]);
				msg.addByte(getCipbiaElement(indexToCombatType(i)));
				msg.addByte(std::max<int8_t>(-100, std::min<int8_t>(100, 100 - damageReduction[i])));
				++combats;
			}
		}
```